src/codegen/memory.rs

```diff
@@ -17,6 +17,8 @@ use crate::codegen::function::compile_function;
 use crate::codegen::type_conversions::wasm_func_type_to_llvm_type;
 use crate::codegen::ModuleCtx;
 
+const MAX_WEBASSEMBLY_PAGES: u32 = 65536;
```

Comment:
Nit: Can we compute this value? Like 2^32 / WASM_PAGE_SIZE? That might be slightly more readable

Reply:
I changed it to `const MAX_WEBASSEMBLY_PAGES: u32 = u32::pow(2, 16);` to match the way it's expressed in the spec. That work for you?